tests/integration/tls/test_tls.py

```diff
@@ -154,3 +157,72 @@ async def test_tls_renewal(ops_test: OpsTest) -> None:
         updated_certs["http_certificates_list"][0]["not_before"]
         > current_certs["http_certificates_list"][0]["not_before"]
     )
+
+
+@pytest.mark.runner(["self-hosted", "linux", "X64", "jammy", "large"])
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_tls_expiration(ops_test: OpsTest) -> None:
+    """Test that expiring TLS certificates are renewed."""
+    # before we can run this test, need to clean up and deploy with different config
+    await ops_test.model.remove_application(APP_NAME, block_until_done=True)
+    await ops_test.model.remove_application(TLS_CERTIFICATES_APP_NAME, block_until_done=True)
+
+    # Deploy TLS Certificates operator
+    config = {"ca-common-name": "CN_CA", "certificate-validity": "1"}
+    await ops_test.model.deploy(TLS_CERTIFICATES_APP_NAME, channel="stable", config=config)
+    await wait_until(ops_test, apps=[TLS_CERTIFICATES_APP_NAME], apps_statuses=["active"])
+
+    # Deploy Opensearch operator
+    await ops_test.model.set_config(MODEL_CONFIG)
+    my_charm = await ops_test.build_charm(".")
+    await ops_test.model.deploy(
+        my_charm,
+        num_units=1,
+        series=SERIES,
+    )
+
+    await wait_until(
+        ops_test,
+        apps=[APP_NAME],
+        units_statuses=["blocked"],
+        wait_for_exact_units=1,
+    )
+
+    # Now apply a hack to make the certificate secrets expire in short time
+    # set the secret expiry to a fixed timedelta of 300 seconds to give time to start initially
```

Comment:
Though technically this is true, I'm not sure if we should go into this level of detail.

From our perspective: we hack the library to create certs with an early expiry.
(Now it's the lib's implementation detail that it goes on to secrtets, etc. but at this level we don't need to know about that.) All we wanted was a cert that expires in 3 mins insteatd the usual, long ahead expiry time.